### PR TITLE
Use new GKE cluster for ci-pr.yaml

### PR DESCRIPTION
**Pull request by NimJay**
### Background
* Currently, the `deployment-tests` GitHub Action is broken — for other pull-requests (see [example error logs](https://github.com/GoogleCloudPlatform/microservices-demo/actions/runs/8254896693/job/22580132965?pr=2406)).
* Specifically, deployment of the `adservice` (Java microservice) is failing:
* The error we're seeing in the GKE cluster:
```
pod/adservice-746b58f97b-gsxzr: container server terminated with exit code 139 ...
A fatal error has been detected by the Java Runtime Environment: ...
Problematic frame:
      > [adservice-746b58f97b-gsxzr server] # C  [profiler_java_agent.so+0x897522]  std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::assign(char const*)+0xc ...
Can not save log file, dump to screen...
VM state: not at safepoint (not fully initialized) ...
VM Mutex/Monitor currently owned by a thread: None ...
```
* Based on our investigation, this seems like an issue with the Nodes currently used by the "online-boutique-prs" GKE cluster. Reason for conclusion: the images deploy fine in other clusters.

### Changes from this PR
* In this PR, we:
   * Add Terraform for a new autopilot GKE cluster ("prs-gke-cluster"). I have `terraform apply`-ed.
   * Update GitHub Actions to use the new GKE cluster instead of the existing "online-boutique-prs" cluster.
   * Add a README.md about using the Terraform.

### How to test
* We have to make sure the GitHub checks work (in this pull-request). Passed. ✅ 
* We should also check the staging URL. Works. ✅ 
* After merging, we should check that:
    * `cleanup.yaml` successfully cleans up the staging deployment created by _this_ pull-request.
    * `ci-main.yaml` successfully deploys the HEAD commit in `main` to the new cluster (prs-gke-cluster).

### Additional info
* Just so we can track the gains from switching to autopilot, I've created a billing report (Jan 1, 2023 to Mar 14, 2024). See [Google-internal report at go/online-boutique-ci-billing-report-2023](http://go/online-boutique-ci-billing-report-2023).
* I have updated [go/sample-app-ci](http://go/sample-app-ci) (Google-internal doc).

### CODE REVIEW

**Review by bourgeoisor — APPROVED** [review]
LGTM. Checks are all passing.

File: .github/workflows/ci-main.yaml [hunk 1]
```diff
@@ -83,8 +83,8 @@ jobs:
       env:
         ACTIONS_ALLOW_UNSECURE_COMMANDS: true
         PROJECT_ID: "online-boutique-ci"
-        PR_CLUSTER: "online-boutique-prs"
-        ZONE: "us-central1-c"
+        PR_CLUSTER: "prs-gke-cluster"
+        REGION: "us-central1"
```
Comment (NimJay) [inline .github/workflows/ci-main.yaml:87]: Since autopilot = regional and standard = zonal.

File: .github/terraform/README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+1. Make sure you have access to the `online-boutique-ci` Google Cloud project.
+1. Move into this folder: `cd .github/terraform`
+1. Set the PROJECT_ID environment variable: `export PROJECT_ID=online-boutique-ci`
+1. Prepare Terraform and download the necessary Terraform dependencies (such as the "hashicorp/google" Terraform provider): `terraform init`
+1. Apply the Terraform: `terraform apply -var project_id=${PROJECT_ID}`
+    * Ideally, you would see `Apply complete! Resources: 0 added, 0 changed, 0 destroyed.` in the output.
+1. Make your desired changes to the Terraform code.
+1. Apply the Terraform: `terraform apply -var project_id=${PROJECT_ID}`
+    * This time, Terraform will prompt you confirm your changes before applying them.
```
Comment (NimJay) [inline .github/terraform/README.md:15]: Ideally, these instructions would be made redundant by a GitHub Action that `terraform apply`s Terraform merged into the `main` branch.

Comment (NimJay) [inline .github/terraform/README.md:15]: Ideally, these instructions would be made redundant by a GitHub Action that `terraform apply`s Terraform merged into the `main` branch.

File: .github/terraform/main.tf [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+  zone    = "us-central1-a"
+}
+
+terraform {
+  # Store the state inside a Google Cloud Storage bucket.
+  backend "gcs" {
+    bucket = "cicd-terraform-state"
+    prefix = "terraform-state"
+  }
+}
```
Comment (NimJay) [inline .github/terraform/main.tf:30]: **Heads-up:** I stored the Terraform state in this bucket.

File: .github/terraform/README.md
Comment (NimJay) [inline .github/terraform/README.md:1]: I've put everything inside the `.github/terraform` folder to match https://github.com/GoogleCloudPlatform/kubernetes-engine-samples.

File: .github/terraform/README.md (hunk 2 above)
Comment (bourgeoisor) [reply .github/terraform/README.md:15]: Interestingly, I have a similar issue on the GKE samples side: https://github.com/GoogleCloudPlatform/kubernetes-engine-samples/issues/611

File: .github/terraform/main.tf [hunk 4]
```diff
@@ -0,0 +1,116 @@
+/**
```
Comment (bourgeoisor) [inline .github/terraform/main.tf:1]: This is such a great idea. I know we have an internal doc with what infra we need to recreate the CICD environment for Online Boutique, but it's nice to have it all codify and easily-reproducible in a TF script. Thanks!

**Review by bourgeoisor — APPROVED** [review]
LGTM!

**Pull request merged**